dependencies.yaml

```diff
@@ -17,31 +40,37 @@ dependencies:
     common:
       - output_types: [conda]
         packages:
-          - cmake>=3.24.0,!=3.30.0
-          - &legate_core legate-core=24.06.*
           - openblas
-          - python-build
+      - output_types: [conda, pyproject, requirements]
+        packages:
+          - cmake>=3.24.0,!=3.30.0
+          - &legate_core legate-core==24.06.*
           - scikit-build>=0.18.0
           - setuptools>=70.0
-  docs:
+  build_tools:
     common:
       - output_types: [conda]
+        packages:
+          - python-build>=1.2.0
+  docs:
+    common:
+      - output_types: [conda, pyproject, requirements]
         packages:
           - sphinx
           - pydata-sphinx-theme
           - myst-parser
   run:
     common:
-      - output_types: [conda]
+      - output_types: [conda, pyproject, requirements]
         packages:
-          - cunumeric=24.06.*
+          - cunumeric==24.06.*
```

Comment:
`pip` doesn't recognized `=`, just `==`. This change is necessary now that this entry is going to end up in `pyproject.toml` too.